src/Meziantou.Analyzer/Rules/OptimizeLinqUsageAnalyzer.cs

```diff
@@ -564,6 +582,53 @@ bool HasTake()
             }
         }
 
+        private static void UseCastInsteadOfSelect(OperationAnalysisContext context, IInvocationOperation operation)
+        {
+            if (!string.Equals(operation.TargetMethod.Name, nameof(Enumerable.Select), StringComparison.Ordinal))
```

Comment:
You should also check that the TargetMethod'scontaining type is `System.Linq.Enumerable`.

Reply:
There's no need, since it is already checked beforehand:
https://github.com/meziantou/Meziantou.Analyzer/blob/81320b9a6d0a8691493414cffb5e9e38500d03bd/src/Meziantou.Analyzer/Rules/OptimizeLinqUsageAnalyzer.cs#L107